plugins/destination/mysql/client/schema.go

```diff
@@ -72,10 +64,6 @@ func (c *Client) getTableColumns(ctx context.Context, tableName string) ([]schem
 		if constraintType != nil {
 			primaryKey = strings.Contains(*constraintType, "PRIMARY KEY")
 		}
-		// subpart only non nil for pks on blob/text columns
-		if subpart != nil && *subpart != maxPrefixLength {
```

Comment:
Not sure why we disable the PK for columns that has a `subpart` but it doesn't match our own number

Reply:
I think this could create issues in a source DB -> dest MySQL sync, for example MySQL -> MySQL